### PR TITLE
Fix(cloud-controller-manager): return error instead of os.Exit when something goes wrong

### PR DESCRIPTION
**Pull request by sanposhiho**
Hi team~

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

The defer function will not be called if os.Exit.
https://play.golang.org/p/f7IjXeb62dd

> deferred functions are not run.
https://golang.org/pkg/os/#Exit

#### Which issue(s) this PR fixes:

part of #102231

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/cloud-controller-manager/main.go [hunk 1]
```diff
@@ -46,6 +47,13 @@ import (
 )
 
 func main() {
+	if err := runCloudControllerManager(); err != nil {
```
Comment (cheftako) [inline cmd/cloud-controller-manager/main.go:50]: Please use a defer.

Comment (khenidak) [reply cmd/cloud-controller-manager/main.go:50]: the error they were trying to solve for is `os.Exit(1)` does not execute defer functions.

**Comment by sanposhiho** [comment]
This PR is not needed anymore because of #105076 :)
So, I close it

**Pull request closed without merging**